RFCs/0043-Fn::FindInMap enhancements (default value and intrinsic functions).md

```diff
@@ -0,0 +1,571 @@
+# Fn::FindInMap enhancements (default value and intrinsic functions)
+
+* **Original Author(s):**: @minggon
+* **Tracking Issue**: [Tracking Issue 43](https://github.com/aws-cloudformation/cfn-language-discussion/issues/43) [Tracking Issue 73](https://github.com/aws-cloudformation/cfn-language-discussion/issues/73) [Tracking Issue 79](https://github.com/aws-cloudformation/cfn-language-discussion/issues/79)
+
+# Summary
+
+`Fn::FindInMap` provides flexibility to customers looking to generalize templates across different attributes that can be specified with parameters. After declaring appropriate key-value mappings in the `Mappings` section, users can supply a `MapName`, `TopLevelKey`, and `SecondLevelKey` to `Fn::FindInMap` to get the corresponding value from the `Mappings` section in the template. 
+
+We will add support for using other intrinsic functions, including `Fn::Select`, `Fn::Join`, `Fn::Sub`, and `Fn::If` within `Fn::FindInMap`. We will also add an optional default value for `Fn::FindInMap`, which will be the output of the `Fn::FindInMap` if the mapping is not present. The first three parameters of `Fn::FindInMap` will remain the same as before, while the optional fourth parameter `DefaultValue` will be of type `String` or `List`.
+
+# Motivation
+
+## Use of other intrinsic functions:
+
+CloudFormation users may want to use intrinsic functions such as `Fn::Select`, `Fn::Join`, `Fn::Sub`, and `Fn::If` within `Fn::FindInMap`. While CloudFormation users can currently nest `Fn::Ref` and `Fn::FindInMap` in `Fn::FindInMap`, more function support has been requested to give CloudFormation users more flexibility when using `Fn::FindInMap`.
```

Comment:
~~I think we should be explicit about which functions are supported here. Currently one of the example uses `Fn::Split`, but that one is not in this list.~~ This is made explicit at the end

A note about current and future functions provided by the transform could be useful too

Reply:
Good point, I think we should be more explicit about what functions are supported under what conditions. Maybe a dedicated section/table listing out all of them(functions and some exceptions such as the `Fn::Select` one) would help?